conda/base/context.py

```diff
@@ -1858,6 +1873,10 @@ def description_map(self):
 
 def reset_context(search_path=SEARCH_PATH, argparse_args=None):
     global context
+
+    # reset plugin config params
+    remove_all_plugin_settings()
+
     context.__init__(search_path, argparse_args)
     context.__dict__.pop("_Context__conda_build", None)
     from ..models.channel import Channel
```

Comment:
I know this isn't part of this PR, but I see the circular dependency comment below and I'm wondering if we should look into improving that issue. At the very least, the comment about importing to avoid a circular dependency is a few lines after the import is done, and that seems a little confusing/easy to miss.

Reply:
@schuylermartin45,

Figuring out how to rework the organization of the code in conda is a right of passage for most new developers on the project. I encourage you to check out my [Honda project](https://github.com/travishathaway/honda) for my own attempt at doing so.

Please use your fresh eyes and energy to help move the project forward!

Reply:
Lol good name on that project. Like I said, at the very least, I think the comment should be bumped up to be above the import so the next time someone goes _WTF_ it'll be a lot easier to find :)